editor/src/components/canvas/controls/breadcrumb-trail.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react'
+import { useEditorState } from '../../../components/editor/store/store-hook'
+import { ElementPathElement } from '../../../components/inspector/sections/header-section/element-path'
+import {
+  betterReactMemo,
+  useKeepReferenceEqualityIfPossible,
+} from '../../../utils/react-performance'
+import { Utils } from '../../../uuiui-deps'
+import * as TP from '../../../core/shared/template-path'
+import { getOpenUtopiaJSXComponentsFromState } from '../../editor/store/editor-state'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { selectComponents } from '../../../components/editor/actions/action-creators'
+import { FlexRow, Icons } from '../../../uuiui'
+
+export const BreadcrumbTrail = () => {
```

Comment:
Please use `betterReactMemo` on all components you define. It improves performance and also gives them a nice label to make debugging easier! (e.g. https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/canvas/canvas-component-entry.tsx#L21)

Reply:
commit #c5a34d1 addresses this